test/test_hmc.py

```diff
@@ -31,3 +32,24 @@ def test_hmc_diag_repr() -> None:
     draws_2 = np.array([hmc_2.sample()[0] for _ in range(M)])
 
     np.testing.assert_array_equal(draws_1, draws_2)
+
+
+def test_hmc_binom() -> None:
+    model = Binomial()
+    M = 800
+    hmc = HMCDiag(
+        model, stepsize=0.08, steps=3, init=np.array([model.initial_state(0)])
+    )
+
+    draws = model.constrain_draws(np.array([hmc.sample()[0] for _ in range(M)]))
+
+    # skip 100 draws to try to make estimates less noisy. e.g treat as "burn in"
+    mean = draws[100:].mean(axis=0)
+    var = draws[100:].var(axis=0, ddof=1)
+
+    print(f"{draws[1:10]=}")
```

Comment:
Is the idea here that this only prints out when tests fail?  If so, I'm OK with that.  otherwise, I'd rather get rid of the test prints.

Reply:
Yes, pytest swallows output and displays it if things fail. I find these print statements have been useful to determine if a failure is something to worry about or not since these are still random tests